src/geometry/narrow_phase.rs

```diff
@@ -301,16 +303,35 @@ impl NarrowPhase {
             let rb1 = &bodies[co1.parent];
             let rb2 = &bodies[co2.parent];
 
-            if (rb1.is_sleeping() || rb1.is_static()) && (rb2.is_sleeping() || rb2.is_static()) {
-                // No need to update this contact because nothing moved.
+            if (rb1.is_sleeping() && rb2.is_static()) || (rb2.is_sleeping() && rb1.is_static()) {
+                // No need to update this proximity because nothing moved.
                 return;
             }
```

Comment:
This is the source of the performance regression when objects are resting.
With our change, two sleeping dynamic bodies will no longer be filtered-out by this test.